### PR TITLE
81 obtain required steps count

### PR DESCRIPTION
**Pull request by otselnik**
If continue transaction to solana fails to execute on solana side because of instruction limit reduse step count to fit in limits.

### CODE REVIEW

File: proxy/plugin/solana_rest_api_tools.py [hunk 1]
```diff
@@ -338,15 +339,33 @@ def send_measured_transaction(client, trx, acc):
...
+            trx.add(TransactionInstruction(program_id=evm_loader_id,
+                                data=bytearray.fromhex("0A") + step_count.to_bytes(8, byteorder='little'),
+                                keys= accounts))
+
+            print("Step count {}", step_count)
+            try:
+                result = send_measured_transaction(client, trx, acc)
+            except SendTransactionError as err:
+                print(err.result['message'])
+                if err.result['message'].find("Program failed to complete") >= 0:
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api_tools.py]: The program can fail due to various reasons. Here it is necessary to highlight the one that is associated with the excess of the number of instructions.

Comment (otselnik) [reply proxy/plugin/solana_rest_api_tools.py]: fixed

File: proxy/plugin/solana_rest_api_tools.py [hunk 2]
```diff
@@ -339,20 +339,44 @@ def send_measured_transaction(client, trx, acc):
...
+                sol_instr_12_cancel(acc, client, accounts)
+                raise
+            except Exception:
+                sol_instr_12_cancel(acc, client, accounts)
+                raise
+            else:
+                break
+
+        if step_count < 1:
+            raise Exception
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api_tools.py]: It's too bad to raise Exception without description. How can you understand the reason if you have this error in CI?

File: proxy/plugin/solana_rest_api_tools.py [hunk 3]
```diff
@@ -339,20 +339,44 @@ def send_measured_transaction(client, trx, acc):
...
         result = object()
-        try:
-            result = send_measured_transaction(client, trx, acc)
-        except Exception:
-            sol_instr_12_cancel(acc, client, accounts)
-            raise
+
+        step_count = initial_step_count
+
+        while step_count > 0:
```
Comment (s-medvedev) [inline proxy/plugin/solana_rest_api_tools.py]: Unclear logic. This code can be simpler. Something like this:
```
def sol_instr_10_continue(acc, client, step_count, accounts):
    while step_count > 0:
       trx = Transaction()
       ...
       try:
         result = send_measured_transaction(client, trx, acc)
         return result
       except SendTransactionError as err:
          if check_if_programm_exceeded_instructions(err.result):
             step_count = int(step_count * 90 / 100)
          else: raise
   raise Exception("Can't execute even one EVM instruction")

def execute_transaction(acc, client, step_count, account):
   try:
      while(True):
         signature = sol_instr_10_continue(acc, client, step_count, account)
         if <execution done>: return signature
   except Exception as err:
      sol_instr_12_cancel(acc, client, accounts)
      raise
```

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**